### ISSUE TITLE
Bug when using specific schema

### ISSUE THREAD

**Issue by MikeKutz** [oddgen/oddgen#18]
If I install a package in the schema 'TEPLSQL', the "name" and "description" of the generator do not display.  This also affects the list of `object_types`.

If I install the same package in a different schema, the values are correct.

My setup
- Oracle 11g (11.2.0.2) XE on Windows 7 x64
- Latest pull of oddgen (and teplsql)
- SQL*Developer 4.1.3.20-78 (latest)
- Packages tested include
  - example 1:1 View in oddgen branch `master`
  - the added package in my fork of oddgen, branch `experimental-DD`

Image of Generator list connected as a user != 'TEPLSQL' shows 3 generators.
1. the first generator is installed in the current connection ( user= 'BOBBY_TABLE')
2. the second generator is installed in the schema 'MKUTZ' (this test the ability to identify generators in different schemas)
3. the third generator is the same code as the 1st generator but is installed in the schema 'TEPLSQL'

![generator list](https://cloud.githubusercontent.com/assets/14933751/15098201/d350eaf4-1504-11e6-8652-d41f755589ce.jpg)

Image of Logging window showing call to "get_name()" is correct for one schema but is NULL if the schema is 'TEPLSQL'
![log file](https://cloud.githubusercontent.com/assets/14933751/15098194/8d57bafa-1504-11e6-867f-c62f573d2742.jpg)

**Comment by PhilippSalvisberg** [comment]
Could you please provide the package spec and body so that I can reproduce the behaviour? Thank you.

**Comment by PhilippSalvisberg** [comment]
To reproduce the behaviour I've dropped all existing generators in my database and installed [extended_view.pks](https://github.com/oddgen/oddgen/blob/master/examples/tutorial/02_extended_view/extended_view.pks) and [extended_view.pkb](https://github.com/oddgen/oddgen/blob/master/examples/tutorial/02_extended_view/extended_view.pkb) in the schema teplsql. Additionally I ran `grant execute on extended_view to public;` to ensure the generator is visible when connected as another user.

In SQL Developer this looks as follows:

![2016-05-14_12-12-05](https://cloud.githubusercontent.com/assets/3883171/15267849/c4975f34-19cd-11e6-83b5-860a3f7531d3.png)

![2016-05-14_12-12-35](https://cloud.githubusercontent.com/assets/3883171/15267850/c4aace84-19cd-11e6-83eb-1c20e57f0879.png)

Connected as `SCOTT` the name is displayed correctly, but when connected as `TEPLSQL` I do not get the name. It's worse, the generator does not work at all and produces the following result:

```
Failed to generate code for TABLE.TE_TEMPLATES via TEPLSQL.EXTENDED_VIEW. Got the following error: ORA-06550: line 2, column 21:
PLS-00302: component 'EXTENDED_VIEW' must be declared
ORA-06550: line 2, column 13:
PL/SQL: Item ignored
ORA-06550: line 5, column 22:
PLS-00302: component 'EXTENDED_VIEW' must be declared
ORA-06550: line 5, column 4:
PL/SQL: Statement ignored
```

The following statement is sent to the database according the logs (you've highlighted that line as well):

```
BEGIN
   ? := TEPLSQL.EXTENDED_VIEW.get_name();
END;
```

The question mark (?) is a JDBC bind variable. Let's rewrite this anonymous block so that it may be executed in a SQL Developer worksheet:

```
DECLARE
   l_name VARCHAR2(1000);
BEGIN
   l_name := TEPLSQL.EXTENDED_VIEW.get_name();
END;
/
```

In SQL Developer worksheet I get the following result when connected as user `TEPLSQL`:

```
Error report -
ORA-06550: line 4, column 22:
PLS-00302: component 'EXTENDED_VIEW' must be declared
ORA-06550: line 4, column 4:
PL/SQL: Statement ignored
06550. 00000 -  "line %s, column %s:\n%s"
*Cause:    Usually a PL/SQL compilation error.
*Action:
```

The very same anonymous block completes successfully when connected as user `SCOTT`. 

This different behaviour is based on the way the PL/SQL parser treats the token `TEPLSQL`. When connected as user `SCOTT` its a schema name, but when connected as user `TEPLSQL` a PL/SQL package named `TEPLSQL` exists and for some reason unknown to me, the parser decides to use this package instead of the schema name and this leads to the error message shown above.

We have several options to solve this problem
1. open an SR at Oracle 
   From my point of view it's a bug, but I might be wrong. Even if it is accepted as a bug, it won't be available soon an for sure not backported to the Express Edition you are using.
2. rename user `TEPLSQL`
   You cannot really rename the user, but you can create a new user and copy/recreate all objects. This way we avoid the conflict with schema and package name.
3. rename the package `TEPLSQL`
   You could rename the package to `TEPLSQL_API` similar to `FTLDB_API`. That's another way to avoid the conflict between schema and package name. But since you do not maintain this package yourself you have to rename the package again whenever you decide to install a new version of tePSQL.
4. do not store generators in schema `TEPLSQL`
   This way you separate the template software installation from you generators and may independently deploy them. The oddgen examples are installed exactly this way. 

Personally I'd go with option 4. However, I do not consider this an oddgen for SQL Developer bug and will close this issue without further actions.

**Comment by MikeKutz** [comment]
Thank you for your work.

The observations makes sense now.

**Comment by PhilippSalvisberg** [comment]
I've opened an SR with the title "Using schema name as package name leads to PL/SQL compile error ORA-06508" on May 14, 2016. Today, I've got the final answer. 

> This is not a bug at all. 

The following links have been mentioned:
- [Qualified Names and Dot Notation](http://docs.oracle.com/database/121/LNPLS/nameresolution.htm#LNPLS01702)
- [Differences Between PL/SQL and SQL Name Resolution Rules](http://docs.oracle.com/database/121/LNPLS/nameresolution.htm#LNPLS99882)

Therefore I assume that the name resolution behaviour of PL/SQL will not be improved in the near future.
